### ISSUE TITLE
Will Play! ever support PSX backwards compatibility? 

### ISSUE THREAD

**Issue by Nobbs66** [jpd002/Play-#327]
With so little documentation on how PSX games ran on the PS2, I'm wondering if it will ever be possible to get it working.

**Comment by jpd002** [comment]
I'm very curious about this too. The IOP doesn't seem to implement the GTE instructions, so, my guess is that they might be emulated by trapping illegal instruction exceptions (or something similar). As for GPU output, I've got no idea how that might be working.

As for actually implementing this in Play!, well, I think that would be a very low priority thing since it wouldn't be very much different from using a PSX emulator.

**Comment by Nobbs66** [comment]
I know I shouldn't normally link PCSX2 threads, but this is the best info I've seen on the subject thus far

http://forums.pcsx2.net/Thread-PS-one-game-compatibility

**Comment by johnnyleap** [comment]
Two segregated OpenGL contexts can be used for rendering; one for PSX output and one for PS2.

Also, the PSX's graphics can be emulated and output without HW acceleration and wouldn't be too demanding on your average CPU today (and upscaling could be possible this way too).

Basically, PS2 emulation can be canceled out once PS1 emulation is executing, at least somewhat -- but I can think of at least three different ways on how this could play out:

1.Play! could try and kind of emulate all PS1 hardware on itself (like writing a PS1 emulator for it) and use the GS for output (as some real PS2 models do); this might be harder than anticipated though.

2.Play! could implement an ad hoc, fully designated PS1 emulator and execute that when a PS1 game is detected, but this is more so like having two emulator cores and picking the correct one per game.

3.Play! could try and mimic how some PS2's really ran PS1 games, which was basically checking what game was on with the firmware, and transferring control to the IOP/MIPS 3000 and then that would have to somehow communicate with the GS and accurately send output. I have no idea how this worked, but it's possible that the GS could mimic the CLUT, textures, primitive, and video output modes to the framebuffer that rendered output just like how the PS1's dedicated chip would.

I agree with it being low priority, but it would be pretty impressive because not only would it be accurately representing PS2, but it will basically be a high-level PS1/PS2 emulator converged in one, needing no BIOS or anything. Quite a few PS1 emulators can just HLE the BIOS anyways and it's honestly far less complex than doing it for PS2.

**Comment by Nobbs66** [comment]
I hear that testing was done on the IOP to check GTE ops. It's great to know that they didn't cut out that part of the R3000 for the PS2. It should make PSX compatibility a lot easier

**Comment by jpd002** [comment]
@Nobbs66 Yeah, that's pretty cool, it means we can have a single emulation core for both the PS2 and the PSX. Still a bit curious about how the PSX's GPU is handled though: no idea if it's directly connected to the CRT output circuit or if the GPU's output is redirected to the GS in some way.

**Comment by Nobbs66** [comment]
I believe one of the PCSX2 devs said it was passed through the GS. Sadly I don't believe much was said beyond that. It seems a lot of testing on actual hardware will have to be done to actually know the details of what's going on.

**Comment by atomic83GitHub** [comment]
The GTE and the MDEC are not emulated, they are integrated in the PS2 console (not sure if it's in the EE or the IOP). As for the PSX gpu, if I remember, it is emulated by the EE+GS.
